### ISSUE TITLE
`tril_indices` and `triu_indices` have strange behavior when `row=0`

### ISSUE THREAD

**Issue by TestSomething22** [pytorch/pytorch#72900]
### 🐛 Describe the bug

`tril_indices` and `triu_indices` have strange behavior when `row=0`. `tril_indices` will return `[0, 0]` as the index but it is wrong since 0 is out of index when `row=0`. Besides, `triu_indices` will raise an error that `Trying to create tensor with negative dimension`.

```python
import torch
results = dict()
row = 0
col = 1
offset = 2
try:
  results["res_1"] = torch.tril_indices(row, col, offset=offset)
except Exception as e:
  results["err_1"] = str(e)
try:
  results["res_2"] = torch.triu_indices(row,col,offset=offset)
except Exception as e:
  results["err_2"] = str(e)

print(results)
```

By contrast, when `row=1` and `col=0`, they both return empty tensor which I think is correct.

```python
import torch
results = dict()
row = 1
col = 0
offset = 2
try:
  results["res_1"] = torch.tril_indices(row, col, offset=offset)
except Exception as e:
  results["err_1"] = "ERROR:"+str(e)
try:
  results["res_2"] = torch.triu_indices(row,col,offset=offset)
except Exception as e:
  results["err_2"] = "ERROR:"+str(e)

print(results)
# {'res_1': tensor([], size=(2, 0), dtype=torch.int64), 'res_2': tensor([], size=(2, 0), dtype=torch.int64)}
```

### Versions

pytorch: 1.10.1

**Comment by zou3519** [comment]
Marking hi-pri for silent correctness issue. Btw numpy.tril_indices is different from torch.tril_indices, but I wasn't sure what else to label this as.

**Comment by ezyang** [comment]
For reference, the output of the script on 1.7.1 is

```
{'res_1': tensor([[0],
        [0]]), 'err_2': 'Trying to create tensor with negative dimension -1: [2, -1]'}
```

**Comment by ezyang** [comment]
And yeah, empty tensors SGTM. Doesn't seem like it should be hard to fix.
